Vostok.Hosting/VostokHost_Extensions.cs

```diff
@@ -47,5 +47,23 @@ public static VostokHost WithConsoleCancellation([NotNull] this VostokHost vosto
 
             return vostokHost;
         }
+        
+        /// <summary>
+        /// Listen <see cref="AppDomain.CurrentDomain.ProcessExit"/> and shutdown vostok if SIGTERM received.
+        /// </summary>
+        public static VostokHost WithSigtermCancellation([NotNull] this VostokHost vostokHost)
+        {
+            AppDomain.CurrentDomain.ProcessExit += (sender, e) =>
+            {
+                vostokHost.ShutdownTokenSource.Cancel();
+                
+                // NOTE: ProcessExit doesn't have flag similar to CancelKeyPress to disable process killing.
+                // NOTE: We have to wait until everything is shut down.
+                // NOTE: We can't use application state there because it is changed before disposing of environment.
+                while(!vostokHost.ApplicationClosed) {}
```

Comment:
This will essentially attempt to consume a whole CPU core. Imagine what would happen if many apps started to shut down at once (this is what actually happens when we drain a Houston host for maintenance).

Wouldn't it work if you just called `Stop` on `VostokHost` and waited for its completion?